x-pack/plugins/security_solution/server/lib/detection_engine/routes/signals/query_signals_route.ts

```diff
@@ -50,19 +50,26 @@ export const querySignalsRoute = (router: SecuritySolutionPluginRouter, config:
           body: '"value" must have at least 1 children',
         });
       }
-      const clusterClient = context.core.elasticsearch.legacy.client;
+      const esClient = context.core.elasticsearch.client.asCurrentUser;
       const siemClient = context.securitySolution!.getAppClient();
 
       // TODO: Once we are past experimental phase this code should be removed
       const { ruleRegistryEnabled } = parseExperimentalConfigValue(config.enableExperimental);
 
       try {
-        const result = await clusterClient.callAsCurrentUser('search', {
+        const { body } = await esClient.search({
           index: ruleRegistryEnabled ? DEFAULT_ALERTS_INDEX : siemClient.getSignalsIndex(),
-          body: { query, aggs, _source, track_total_hits, size },
-          ignoreUnavailable: true,
+          body: {
+            query,
+            // Note: I use a spread operator to please TypeScript with aggs: { ...aggs }
```

Comment:
👍 🙇

Reply:
Yeah, I was trying to be honest around that spread operator to get around TypeScript issues.